client.go

```diff
@@ -1881,6 +1881,67 @@ func (c *HostClient) cachedTLSConfig(addr string) *tls.Config {
 	return cfg
 }
 
+// AcquireHostClient returns an empty HostClient instance from host client pool.
+//
+// The returned HostClient instance may be passed to ReleaseHostClient when it is
+// no longer needed. This allows HostClient recycling, reduces GC pressure
+// and usually improves performance.
+func AcquireHostClient() *HostClient {
+	v := hostClientPool.Get()
+	if v == nil {
+		return &HostClient{}
+	}
+	return v.(*HostClient)
+}
+
+// ReleaseHostClient returns hc acquired via AcquireHostClient to host client pool.
+//
+// It is forbidden accessing hc and/or its' members after returning
+// it to host client pool.
+func ReleaseHostClient(hc *HostClient) {
+	hc.Reset()
+	hostClientPool.Put(hc)
+}
+
+// Reset clears host client contents.
+func (hc *HostClient) Reset() {
+	hc.Addr = ""
+	hc.Name = ""
+	hc.Dial = nil
+	hc.DialDualStack = false
+	hc.IsTLS = false
+	hc.TLSConfig = nil
+	hc.MaxConns = 0
+	hc.MaxConnDuration = 0
+	hc.MaxIdleConnDuration = 0
+	hc.MaxIdemponentCallAttempts = 0
+	hc.ReadBufferSize = 0
+	hc.WriteBufferSize = 0
+	hc.ReadTimeout = 0
+	hc.WriteTimeout = 0
+	hc.MaxResponseBodySize = 0
+	hc.DisableHeaderNamesNormalizing = false
+	hc.DisablePathNormalizing = false
+	hc.MaxConnWaitTimeout = 0
+	hc.RetryIf = nil
+	hc.clientName = atomic.Value{}
+	hc.lastUseTime = 0
+	hc.connsCount = 0
+	hc.conns = hc.conns[:0]
+	if hc.connsWait != nil {
+		hc.connsWait.head = hc.connsWait.head[:0]
+		hc.connsWait.headPos = 0
+		hc.connsWait.tail = hc.connsWait.tail[:0]
+	}
```

Comment:
I'm wondering if we should do something here when `hc.connsCount` isn't `0`. It means there are still open connections and just resetting all fields means they can't be cleaned up and closed anymore.

We could close all connections here. And we would also have to make sure no goroutine is waiting on `connsWait` anymore.
The alternative is just panicing as the caller needs to make sure the HostClient isn't in use anymore. If they don't it's a serious flaw in their program and just closing connections might just make that flaw harder to see.

Reply:
My intention is that the user decides whether to release the host, because we have exposed connsCount. Maybe I should make it clearer in the document.

Reply:
I think it's better to also just panic here in case the user didn't wait until `connsCount == 0`. Cause thats a bug you won't easily notice otherwise.

Reply:
You're right. I'll add panic and warning in docs.

Reply:
I rethought `hostClient.ConnsCount()` again. If user use hostClient concurrently, we can not ensure we get the correct connsCount. Any ideas?

Reply:
Why not?